lib/bitcoin/BitcoinProcessor.ts

```diff
@@ -0,0 +1,590 @@
+    const request = {
+      method: 'getblockcount'
+    };
+    const response = await this.rpcCall(request);
+    return response;
+  }
+
+  /**
+   * Given a Bitcoin block height and hash, verifies against the blockchain
+   * @param height Block height to verify
+   * @param hash Block hash to verify
+   * @returns true if valid, false otherwise
+   */
+  private async verifyBlock (height: number, hash: string): Promise<boolean> {
+    console.info(`Verifying block ${height} (${hash})`);
+    const responseData = await this.rpcCall({
+      method: 'getblockbyheight',
+      params: [
+        height,  // height
+        true,   // verbose (block details)
+        false    // details (transaction details)
+      ]
+    });
+
+    console.debug(`Retrieved block ${responseData.height} (${responseData.hash})`);
+
+    let actualHash: string = responseData.hash;
+    return hash === actualHash;
+  }
+
+  /**
+   * Given a Bitcoin block height, processes that block for Sidetree transactions
+   * @param block Block height to process
+   * @returns the block hash processed
+   */
+  private async processBlock (block: number): Promise<string> {
+    console.info(`Processing block ${block}`);
+    const responseData = await this.rpcCall({
+      method: 'getblockbyheight',
+      params: [
+        block,  // height
+        true,   // verbose (block details)
+        true    // details (transaction details)
+      ]
+    });
+
+    const transactions = responseData.tx as Array<any>;
+    const blockHash = responseData.hash;
+    let anchorFilePosition = 0;
+
+    // console.debug(`Block ${block} contains ${transactions.length} transactions`);
+
+    // iterate through transactions
+    for (let transactionIndex = 0; transactionIndex < transactions.length; transactionIndex++) {
+      // get the output coins in the transaction
+      if (!('vout' in transactions[transactionIndex])) {
+        // console.debug(`Skipping transaction ${transactionIndex}: no output coins.`);
+        continue;
+      }
+      const outputs = transactions[transactionIndex].vout as Array<any>;
+      for (let outputIndex = 0; outputIndex < outputs.length; outputIndex++) {
+        // grab the scripts
+        const script = outputs[outputIndex].scriptPubKey;
+
+        // console.debug(`Checking transaction ${transactionIndex} output coin ${outputIndex}: ${JSON.stringify(script)}`);
+        // check for returned data for sidetree prefix
+        const hexDataMatches = script.asm.match(/\s*OP_RETURN ([0-9a-fA-F]+)$/);
+        if (!hexDataMatches || hexDataMatches.length === 0) {
+          continue;
+        }
+        const data = Buffer.from(hexDataMatches[1], 'hex').toString();
+        if (data.startsWith(this.sidetreePrefix)) {
+          // we have found a sidetree transaction
+          const sidetreeTransaction: ITransaction = {
+            transactionNumber: TransactionNumber.construct(block, anchorFilePosition),

```

Comment:
I like your idea of using `transactionIndex` instead, you reckon we can?

Reply:
This is the same behavior as the old code. This particular "index" is a counter incremented within this conditional block.